server/journeys/referral/area-of-work/tests.test.ts

```diff
@@ -0,0 +1,154 @@
+import testRequestCaptor from '../../../routes/testutils/testRequestCaptor'
+import { HmppsUser } from '../../../interfaces/hmppsUser'
+import createTestHtmlElement from '../../../routes/testutils/createTestHtmlElement'
+import { JourneyData } from '../../../@types/express'
+
+const uuid = uuidv4()
+let app: Express
+const csipApiService = {
+  getReferenceData: () => [
+    { code: 'A', description: 'TEXT' },
+    { code: 'B', description: 'TEXT2' },
+  ],
+} as unknown as CsipApiService
+const [reqCaptured, requestCaptor] = testRequestCaptor()
+
+beforeEach(() => {
+  app = appWithAllRoutes({
+    services: { csipApiService },
+    requestCaptor,
+  })
+})
+
+afterEach(() => {
+  jest.resetAllMocks()
+})
+
+describe('GET /referral/area-of-work', () => {
+  it('render page', async () => {
+    const result = await request(app).get(`/${uuid}/referral/area-of-work`).expect(200).expect('Content-Type', /html/)
+    const html = createTestHtmlElement(result.text)
+    expect(getByText(html, 'Which area do you work in?')).toBeVisible()
+    expect((getByRole(html, 'option', { name: 'Select area' }) as HTMLOptionElement).defaultSelected).toBeTruthy()
+    expect((getByRole(html, 'option', { name: 'TEXT' }) as HTMLOptionElement).defaultSelected).toBeFalsy()
+    expect((getByRole(html, 'option', { name: 'TEXT2' }) as HTMLOptionElement).defaultSelected).toBeFalsy()
+  })
+
+  it('pre-fill form with values from journeyData', async () => {
+    const appWithJourneyDataInjected = appWithAllRoutes({
+      services: { csipApiService },
+      requestCaptor: testRequestCaptor(
+        { referral: { refererArea: { code: 'A', description: 'TEXT' } } } as JourneyData,
+        uuid,
+      )[1],
+    })
+
+    const result = await request(appWithJourneyDataInjected)
+      .get(`/${uuid}/referral/area-of-work`)
+      .expect(200)
+      .expect('Content-Type', /html/)
+    const html = createTestHtmlElement(result.text)
+    expect((getByRole(html, 'option', { name: 'Select area' }) as HTMLOptionElement).defaultSelected).toBeFalsy()
+    expect((getByRole(html, 'option', { name: 'TEXT' }) as HTMLOptionElement).defaultSelected).toBeTruthy()
+    expect((getByRole(html, 'option', { name: 'TEXT2' }) as HTMLOptionElement).defaultSelected).toBeFalsy()
+  })
+
+  it('render validation errors if any', async () => {
+    const appWithError = appWithAllRoutes({
+      services: { csipApiService },
+      requestCaptor,
+      validationErrors: { areaOfWork: ['Select your area of work'] },
+    })
+
+    const result = await request(appWithError)
+      .get(`/${uuid}/referral/area-of-work`)
+      .expect(200)
+      .expect('Content-Type', /html/)
+
+    const html = createTestHtmlElement(result.text)
+    expect(getByText(html, 'Which area do you work in?')).toBeVisible()
+    expect(getByText(html, 'There is a problem')).toBeVisible()
+    expect(getByText(html, 'Select your area of work')).toBeVisible()
```

Comment:
I liked James' `const error = getByRole(topLevelElement, 'link', { name: onBehalfOfError }) expect(error).toBeVisible()` test for this as it asserts that the error summary links to the correct form element